### DISCUSSION TITLE
Sandboxing / Executing of Untrusted Code

### DISCUSSION THREAD

**Discussion by zicklag** [HigherOrderCO/HVM1#173] [Q&A] [answered]
I'm currently using JavaScript as a scripting language for games, which is great because JS is well sandboxed, and can be used, similar to a Browser web page, to run untrusted mods, without having to worry about somebody being able to take over your machine ( unless of course you add JS bindings to the filesystem, etc. )

I was curious if in the future, you would consider HVM useful in the same scenario, where you may not trust the code you are executing for mods.

I know with such a brand-new project we can't necessarily expect rock solid security yet, but I wanted to know if, in concept anyway, that should be reasonable in a fully mature HVM.

This would also feel _way_ more secure if we managed to pull of a proof of the HVM invariants ( https://github.com/Kindelia/HVM/discussions/120 ), of course, which could make it even _better_ than JavaScript.

**Comment by steinerkelvin** [comment, marked as answer]
In my understanding, this is just a matter of instantiating a clean slate HVM runtime without any of the default IOs and just injecting the relevant rules for the script system making sure they guarantee the sandboxing. And... coming to trust that the VM does not have a remote-code-execution-like vulnerability. Formalizing the HVM would indeed be a very strong step towards the latter.

**Reply by zicklag** [reply]
Cool, that was my understanding. 👍
